[Caffe2] ld: can't map file, errno=22 file '/usr/local/cuda/lib/stubs/cuda.framework' for architecture x86_64

## Issue description
error occurs when linking the cuda library

ld: can't map file, errno=22 file '/usr/local/cuda/lib/stubs/cuda.framework' for architecture x86_64

Provide a short description.

I have tracked down in cuda.cmake

`find_library(CUDA_CUDA_LIB cuda
    PATHS ${CUDA_TOOLKIT_ROOT_DIR}
    PATH_SUFFIXES lib lib64 lib/stubs lib64/stubs lib/x64)`

finds the cuda.framework instead of libcuda.dylib.

Have confirm it was broken by this commit (https://github.com/pytorch/pytorch/commit/4bf0202cac4cdf95bf437df5d2e8a8c5bb892569#diff-e1f6ba8d0726bd4dc7d55fa517bd381c)

I want to contribute, any idea why it is linking the cuda.framework?

## System Info

CUDA used to build PyTorch: 9.2

OS: Mac OSX 10.13.4
GCC version: Could not collect
CMake version: version 3.11.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.2.64
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Probably one of the following:
/usr/local/cuda/lib/libcudnn.7.dylib
/usr/local/cuda/lib/libcudnn_static.a

@pjh5 
[ 94%] Linking CXX shared library ../lib/libcaffe2_gpu.dylib
cd /Users/hongbinliu/pytorch/build/caffe2 && /usr/local/miniconda3/bin/cmake -E cmake_link_script CMakeFiles/caffe2_gpu.dir/link.txt --verbose=1
/Library/Developer/CommandLineTools/usr/bin/clang++   -msse4.1 -msse4.2   --std=c++11  -Wno-deprecated -fvisibility-inlines-hidden -Wno-deprecated-declarations -DONNX_NAMESPACE=onnx_c2 -O2 -fPIC -Wno-narrowing -Wno-invalid-partial-specialization -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-unused-parameter -Wno-unknown-warning-option -Wno-unknown-pragmas -DHAVE_AVX_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -O3  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.10 -dynamiclib -Wl,-headerpad_max_install_names -compatibility_version 1.0.0 -current_version 1.0.0 -o ../lib/libcaffe2_gpu.1.dylib -install_name @rpath/libcaffe2_gpu.1.dylib CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cudnn/AffineGridGenerator.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cudnn/BatchNorm.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cudnn/Conv.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cudnn/GridSampler.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cudnn/RNN.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/cuda/CUDAGenerator.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/cuda/PinnedMemoryAllocator.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/cuda/detail/CUDAHooks.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/CUDAReduceOps.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/CUDAUnaryOps.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAByteStorage.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAByteTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAByteType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDACharStorage.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDACharTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDACharType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDACopy.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDADoubleStorage.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDADoubleTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDADoubleType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAFloatStorage.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAFloatTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAFloatType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAHalfStorage.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAHalfTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAHalfType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAIntStorage.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAIntTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAIntType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDALongStorage.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDALongTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDALongType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAShortStorage.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAShortTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/CUDAShortType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/RegisterCUDA.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDAByteTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDAByteType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDACharTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDACharType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDADoubleTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDADoubleType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDAFloatTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDAFloatType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDAIntTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDAIntType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDALongTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDALongType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDAShortTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/SparseCUDAShortType.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCCachingAllocator.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCCachingHostAllocator.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCGeneral.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCStorageCopy.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCStream.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCTensorCopy.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCTensorRandom.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/THCThreadLocal.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCS/THCSTensor.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/cudnn/Descriptors.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/cudnn/Handles.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/cudnn/Types.cpp.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCReduceApplyUtils.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCBlas.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCSleep.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCStorage.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCStorageCopy.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensor.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorCopy.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorMath.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorMathBlas.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorMathMagma.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorMathPairwise.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorMathReduce.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorMathScan.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorIndex.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorConv.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorRandom.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorScatterGather.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorTopK.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorSort.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorTypeUtils.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCSortUtils.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCTensorMode.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorSortByte.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareTByte.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathPointwiseByte.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareByte.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathReduceByte.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMaskedByte.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorSortChar.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareTChar.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathPointwiseChar.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareChar.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathReduceChar.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMaskedChar.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorSortShort.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareTShort.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathPointwiseShort.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareShort.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathReduceShort.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMaskedShort.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorSortInt.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareTInt.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathPointwiseInt.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareInt.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathReduceInt.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMaskedInt.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorSortLong.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareTLong.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathPointwiseLong.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareLong.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathReduceLong.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMaskedLong.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorSortHalf.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareTHalf.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathPointwiseHalf.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareHalf.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathReduceHalf.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMaskedHalf.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorSortFloat.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareTFloat.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathPointwiseFloat.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareFloat.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathReduceFloat.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMaskedFloat.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorSortDouble.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareTDouble.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathPointwiseDouble.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathCompareDouble.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMathReduceDouble.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/generated/caffe2_gpu_generated_THCTensorMaskedDouble.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THC/caffe2_gpu_generated_THCHalf.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_AbsCriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_Abs.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_BatchNormalization.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_BCECriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_ClassNLLCriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_Col2Im.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_DistKLDivCriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_ELU.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_FeatureLPPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_FusedRNNKernel.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_GatedLinearUnit.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_HardTanh.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_Im2Col.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_IndexLinear.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_L1Cost.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_LeakyReLU.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_LogSigmoid.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_LookupTableBag.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_LookupTable.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_MarginCriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_MSECriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_MultiLabelMarginCriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_MultiMarginCriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_PReLU.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_RReLU.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_Sigmoid.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SmoothL1Criterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SoftMarginCriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SoftPlus.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SoftShrink.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SparseLinear.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialAdaptiveAveragePooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialAdaptiveMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialAveragePooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialClassNLLCriterion.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialConvolutionLocal.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialConvolutionMM.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialCrossMapLRN.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialDepthwiseConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialDilatedConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialDilatedMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialFractionalMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialFullConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialFullDilatedConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialGridSamplerBilinear.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialMaxUnpooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialReflectionPadding.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialReplicationPadding.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialSubSampling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialUpSamplingBilinear.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_SpatialUpSamplingNearest.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_Sqrt.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_Square.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_Tanh.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_TemporalConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_TemporalMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_TemporalReflectionPadding.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_TemporalReplicationPadding.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_TemporalRowConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_TemporalUpSamplingLinear.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_TemporalUpSamplingNearest.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_Threshold.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricAdaptiveAveragePooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricAdaptiveMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricAveragePooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricDilatedConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricDilatedMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricFractionalMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricFullConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricFullDilatedConvolution.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricGridSamplerBilinear.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricMaxPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricMaxUnpooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricReplicationPadding.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricUpSamplingNearest.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCUNN/caffe2_gpu_generated_VolumetricUpSamplingTrilinear.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCS/caffe2_gpu_generated_THCSTensor.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/THCS/caffe2_gpu_generated_THCSparse.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/cuda/caffe2_gpu_generated_CUDAHalf.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/cuda/detail/caffe2_gpu_generated_IndexUtils.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_Distributions.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_Embedding.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_EmbeddingBag.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_Gesv.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_RoiPooling.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_SoftMax.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_SparseMM.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_SpectralOps.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_TensorCompare.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_TensorFactories.cu.o CMakeFiles/caffe2_gpu.dir/__/aten/src/ATen/native/cuda/caffe2_gpu_generated_Unique.cu.o -Wl,-rpath,/usr/local/miniconda3/lib -Wl,-rpath,/usr/local/cuda/lib -Wl,-rpath,/Users/hongbinliu/pytorch/build/lib /usr/local/cuda/lib/libcudart_static.a -Wl,-rpath,/usr/local/cuda/lib /usr/local/miniconda3/lib/libmkl_intel_lp64.dylib /usr/local/miniconda3/lib/libmkl_intel_thread.dylib /usr/local/miniconda3/lib/libmkl_core.dylib /usr/local/opt/libomp/lib/libiomp5.dylib /usr/lib/libpthread.dylib /usr/lib/libm.dylib /usr/local/miniconda3/lib/libirc.dylib /usr/local/miniconda3/lib/libmkl_intel_lp64.dylib /usr/local/miniconda3/lib/libmkl_intel_thread.dylib /usr/local/miniconda3/lib/libmkl_core.dylib /usr/local/opt/libomp/lib/libiomp5.dylib /usr/lib/libpthread.dylib /usr/lib/libm.dylib /usr/local/miniconda3/lib/libirc.dylib /usr/local/cuda/lib/libcudart_static.a -Wl,-rpath,/usr/local/cuda/lib /usr/local/cuda/lib/libcusparse.dylib /usr/local/cuda/lib/libcurand.dylib ../lib/libcaffe2.1.dylib /usr/local/cuda/lib/stubs/cuda.framework /usr/local/cuda/lib/libcurand.dylib /usr/local/cuda/lib/libnvrtc.dylib /usr/local/cuda/lib/libcudnn.7.dylib /usr/local/miniconda3/lib/libmkl_intel_lp64.dylib /usr/local/miniconda3/lib/libmkl_intel_thread.dylib /usr/local/miniconda3/lib/libmkl_core.dylib /usr/local/opt/libomp/lib/libiomp5.dylib /usr/lib/libpthread.dylib /usr/lib/libm.dylib /usr/local/miniconda3/lib/libirc.dylib /usr/local/cuda/lib/libcufft.dylib /usr/local/cuda/lib/libcublas.dylib /usr/local/cuda/lib/libcublas_device.a
ld: can't map file, errno=22 file '/usr/local/cuda/lib/stubs/cuda.framework' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libcaffe2_gpu.1.dylib] Error 1
make[1]: *** [caffe2/CMakeFiles/caffe2_gpu.dir/all] Error 2
make: *** [all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,6 +30,15 @@ if(NOT DEFINED BLAS_SET_BY_USER)
   set(BLAS_SET_BY_USER ${BLAS_SET_BY_USER} CACHE STRING "Marks whether BLAS was manually set by user or auto-detected")
 endif()
 
+# These lines are an attempt to make find_package(cuda) pick up
+# libcuda.dylib, and not cuda.framework.  It doesn't work all
+# the time, but it seems to help for some users.
+# TODO: replace this with a more robust fix
+if(APPLE)
+  set(CMAKE_FIND_FRAMEWORK LAST)
+  set(CMAKE_FIND_APPBUNDLE LAST)
+endif()
+
 # ---[ Options.
 # Note to developers: if you add an option below, make sure you also add it to
 # cmake/Summary.cmake so that the summary prints out the option values.

--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -399,7 +399,20 @@ if(USE_CUDA)
     # caffe2::cudart is dealt with separately, due to CUDA_ADD_LIBRARY
     # design reason (it adds CUDA_LIBRARIES itself).
     set(Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS
-        caffe2::cuda caffe2::cufft caffe2::curand caffe2::nvrtc)
+        caffe2::cufft caffe2::curand)
+    if(BUILD_CAFFE2)
+      # Don't be deceived!  caffe2::cuda is the low-level DRIVER API,
+      # not the actual CUDA library.  Caffe2 depends directly on nvrtc
+      # and needs it, but ATen doesn't use it at all, and so the
+      # dependency is unnecessary.
+      #
+      # BTW, if you change this so that PyTorch has this dependency
+      # again, make sure Mac OS X GPU builds still work; there's
+      # a decent chance the library finding algorithm picked up
+      # on cuda.framework, which is totally not going to work when
+      # linking.
+      list(APPEND Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS caffe2::cuda caffe2::nvrtc)
+    endif()
     if(CAFFE2_FOUND_CUDNN)
       LIST(APPEND Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS caffe2::cudnn)
     else()

